crates/contracts/src/lib.rs

```diff
@@ -124,29 +128,38 @@ mod tests {
             }};
         }
 
-        for network in &[1, 4, 100] {
+        for network in &[MAINNET, GOERLI, GNOSIS] {
             assert_has_deployment_address!(GPv2Settlement for *network);
             assert_has_deployment_address!(SushiSwapFactory for *network);
             assert_has_deployment_address!(SushiSwapRouter for *network);
             assert_has_deployment_address!(WETH9 for *network);
+            assert_has_deployment_address!(CowProtocolToken for *network);
+            assert_has_deployment_address!(CowProtocolVirtualToken for *network);
         }
-        for network in &[1, 4] {
+        for network in &[MAINNET, GOERLI] {
             assert_has_deployment_address!(BalancerV2Vault for *network);
-            assert_has_deployment_address!(BalancerV2LiquidityBootstrappingPoolFactory for *network);
-            assert_has_deployment_address!(BalancerV2NoProtocolFeeLiquidityBootstrappingPoolFactory for *network);
             assert_has_deployment_address!(BalancerV2WeightedPoolFactory for *network);
             assert_has_deployment_address!(BalancerV2WeightedPool2TokensFactory for *network);
-            assert_has_deployment_address!(BalancerV2StablePoolFactory for *network);
             assert_has_deployment_address!(UniswapV2Factory for *network);
             assert_has_deployment_address!(UniswapV2Router02 for *network);
+            assert_has_deployment_address!(UniswapV3SwapRouter for MAINNET);
+            assert_has_deployment_address!(IUniswapV3Factory for MAINNET);
         }
-        #[allow(clippy::single_element_loop)]
-        for network in &[100] {
-            assert_has_deployment_address!(HoneyswapFactory for *network);
-            assert_has_deployment_address!(HoneyswapRouter for *network);
-        }
-        assert_has_deployment_address!(BalancerV2StablePoolFactoryV2 for 1);
-        assert_has_deployment_address!(UniswapV3SwapRouter for 1);
+
+        // only gnosis
+        assert_has_deployment_address!(BaoswapFactory for GNOSIS);
+        assert_has_deployment_address!(BaoswapRouter for GNOSIS);
+        assert_has_deployment_address!(HoneyswapFactory for GNOSIS);
+        assert_has_deployment_address!(HoneyswapRouter for GNOSIS);
+        assert_has_deployment_address!(SwaprFactory for GNOSIS);
+        assert_has_deployment_address!(SwaprRouter for GNOSIS);
+
+        // only mainnet
+        assert_has_deployment_address!(BalancerV2StablePoolFactory for MAINNET);
+        assert_has_deployment_address!(BalancerV2StablePoolFactoryV2 for MAINNET);
+        assert_has_deployment_address!(BalancerV2LiquidityBootstrappingPoolFactory for MAINNET);
+        assert_has_deployment_address!(BalancerV2NoProtocolFeeLiquidityBootstrappingPoolFactory for MAINNET);
```

Comment:
Just to double check `BalancerV2NoProtocolFeeLiquidityBootstrappingPoolFactory` was available on `rinkeby` but is no longer available on `goerli`?

Reply:
Goerli has a very limited set of pool factories AFAIU.

Reply:
No `BalancerV2NoProtocolFeeLiquidityBootstrappingPoolFactory ` on `goerli`